mm2src/coins/utxo/rpc_clients/electrum_rpc/connection.rs

```diff
@@ -0,0 +1,872 @@
+                secure_connection = true;
+                format!("wss://{address}")
+            },
+            ElectrumProtocol::TCP | ElectrumProtocol::SSL => {
+                disconnect_and_return!(
+                    Irrecoverable,
+                    "'TCP' and 'SSL' are not supported in a browser. Please use 'WS' or 'WSS' protocols",
+                    connection,
+                    event_handlers
+                );
+            },
+        };
+
+        // Try to connect to the server.
+        let now = Instant::now();
+        let Ok(connect_f) = ws_transport(
+            CONN_IDX.fetch_add(1, AtomicOrdering::Relaxed),
+            &protocol_prefixed_address,
+            &connection.weak_spawner()
+        )
+        .boxed()
+        .timeout_secs(timeout)
+        .await else {
+            disconnect_and_return!(
+                ElectrumConnectionErr::Timeout(timeout),
+                connection,
+                event_handlers
+            );
+        };
+        // We will use the remaining timeout for version checking.
+        let timeout = (timeout - now.elapsed().as_secs_f64()).max(0.0);
+
+        let (mut transport_tx, mut transport_rx) =
+            disconnect_and_return_if_err!(connect_f, Temporary, connection, event_handlers);
+
+        match secure_connection {
+            true => info!("Electrum client connected to {address} securely"),
+            false => info!("Electrum client connected to {address}"),
+        };
+
+        let (connection_ready_signal, wait_for_connection_ready) = async_oneshot::channel();
+
+        let connection_loop = {
+            let last_response = Arc::new(AtomicU64::new(now_ms()));
+            let no_connection_timeout_f = {
+                let last_response = last_response.clone();
+                async move {
+                    loop {
+                        Timer::sleep(CUTOFF_TIMEOUT).await;
+                        let last_sec = (last_response.load(AtomicOrdering::Relaxed) / 1000) as f64;
+                        if now_float() - last_sec > CUTOFF_TIMEOUT {
+                            break ElectrumConnectionErr::Temporary(format!(
+                                "Server didn't respond for too long ({}s).",
+                                now_float() - last_sec
+                            ));
+                        }
+                    }
+                }
+            };
+            let mut no_connection_timeout_f = Box::pin(no_connection_timeout_f).fuse();
+
+            let recv_f = {
+                let address = address.clone();
+                let connection = connection.clone();
+                let event_handlers = event_handlers.clone();
+                async move {
+                    // FIXME: `process_electrum_response` would be better off receiving bytes and not raw json,
+                    // this is to avoid converting the json back to bytes to pass to the event handlers. Check if
+                    // this is possible.
```

Comment:
I am pretty sure electrum returns JSON responses only.

Reply:
I mean this:

https://github.com/KomodoPlatform/komodo-defi-framework/blob/76b2f689066a83d8d2d8838c2f33770eed03b16a/mm2src/coins/utxo/rpc_clients/electrum_rpc/connection.rs#L323-L326

Having the response in raw bytes format (before parsing to json) is easier for the `RpcTransportEventHandler`s to deal with since they ingest `&[u8]`.
I remember fiddling with the wasm part to make it spit out raw bytes and not json but failed. will give it another go.

Reply:
Done in https://github.com/KomodoPlatform/komodo-defi-framework/pull/1966/commits/c9b669f611fe69522babb2b914d8867be3b5575f